### ISSUE TITLE
What might be the cause trained models fail to generalize beyond 5 seconds of audio?

### ISSUE THREAD

**Issue by yl4579** [SWivid/F5-TTS#653]
### Checks

- [X] This template is only for question, not feature requests or bug reports.
- [X] I have thoroughly reviewed the project documentation and read the related paper(s).
- [X] I have searched for existing issues, including closed ones, no similar questions.
- [X] I confirm that I am using English to submit this report in order to facilitate communication.

### Question details

I have tried to train the model for a different project and found a weird pattern: the model fails to produce audio after 5 seconds of speech, regardless of the length of the prompt. For example, if the prompt is 1s, the generated audio can be 4s, but beyond that, it becomes gibberish. Similarly, if the prompt is 5s, everything afterward is gibberish. 

Since the hop size is 256, 512 frames are around 5 seconds, so I think it could be caused by either positional embeddings or dataloader. I was using [this commit](https://github.com/SWivid/F5-TTS/commit/32c3ee77017d728321ffaa5d10e9f9f6cd44f20d) on Oct 22, so there might be a bug. Should I upgrade to the latest repo, or has anything changed since then? Did anyone else have similar issues? 

The only similar issue I found was #574 but I wasn't doing fine-tuning.

**Comment by SWivid** [comment]
yep, if train from scratch and the samples in dataset are up to 5 seconds, the inference will also limit to 5 sec.
could use some longer audio samples to train the model

**Comment by yl4579** [comment]
I used the default set up, did it get truncated? How do I disable this?

**Comment by yl4579** [comment]
Also why can’t the architecture generalize to longer duration unseen during training? From my experience, architectures like DiTTo-TTS can but E2/F5 in this repo can’t. Is it caused by the position embeddings?

**Comment by SWivid** [comment]
> I used the default set up, did it get truncated? How do I disable this?

thought only finetune_gradio.py is applying slicer (to disable, #574 )
if using prepare_xxdataset.py and train.py will not do truncating

> Also why can’t the architecture generalize to longer duration unseen during training? From my experience, architectures like DiTTo-TTS can but E2/F5 in this repo can’t. Is it caused by the position embeddings?

e2/f5 is padding with filler tokens, f5 has still a very weak but a little bit stronger generalize-to-longer ability then e2 with text refinement module.
so you have implemented ditto-tts and mel-vae codec? welcome to share if you got some insights on its extrapolation capability

**Comment by SWivid** [comment]
@lumpidu sure. so if no modification done, tha max is limit to 15 sec in code
if train set is up to 20 sec and use a ∞ sec ref_audio, ref_audio will be clipped to 15 sec, and gen_audio support up to 5 sec
